### PR TITLE
Lashemilt/issue152

### PR DESCRIPTION
**Pull request by LAShemilt**
#152 removed dependency on pathtools with the methodolgy suggested in #153

### CODE REVIEW

File: rfi_file_monitor/engines/directory_watchdog_engine.py [hunk 1]
```diff
@@ -122,7 +123,7 @@ def _search_for_existing_files(self, directory: Path) -> int:
         for child in directory.iterdir():
             if child.is_file() \
                 and not child.is_symlink() \
-                and match_path(str(child), included_patterns=self._included_patterns, excluded_patterns=self._excluded_patterns,
+                and match_path(PurePosixPath(child), included_patterns=self._included_patterns, excluded_patterns=self._excluded_patterns,
```
Comment (tschoonj) [inline rfi_file_monitor/engines/directory_watchdog_engine.py:126]: `child` is already an instance of `Path` here, casting won't work here I think. Also, `PurePosixPath` is only appropriate on macOS/Linux.

File: rfi_file_monitor/engines/directory_watchdog_engine.py [hunk 2]
```diff
@@ -133,10 +134,10 @@ def _search_for_existing_files(self, directory: Path) -> int:
     def _dir_filter(self, child: Path) -> bool:
         return child.is_dir() and \
             not child.is_symlink() and \
-            match_path(str(child),
+            match_path(PurePosixPath(child),
```
Comment (tschoonj) [inline rfi_file_monitor/engines/directory_watchdog_engine.py:137]: Same comment as above.

File: rfi_file_monitor/engines/file_watchdog_engine.py [hunk 3]
```diff
@@ -127,7 +127,7 @@ def _search_for_existing_files(self, directory: Path) -> List[RegularFile]:
         for child in directory.iterdir():
             if child.is_file() \
                 and not child.is_symlink() \
-                and match_path(str(child), included_patterns=self._included_patterns, excluded_patterns=self._excluded_patterns,
+                and match_path(PurePosixPath(child), included_patterns=self._included_patterns, excluded_patterns=self._excluded_patterns,
```
Comment (tschoonj) [inline rfi_file_monitor/engines/file_watchdog_engine.py:130]: Same comment as above

File: rfi_file_monitor/file.py [hunk 4]
```diff
@@ -242,7 +242,7 @@ def excluded_patterns(self):
     def _get_filelist(self, _dir: Path) -> List[Tuple[str, int]]:
         rv: List[Tuple[str, int]] = []
         for entry in _dir.iterdir():
-            if not match_path(str(entry), included_patterns=self._included_patterns, excluded_patterns=self._excluded_patterns, case_sensitive=False):
+            if not match_path(PurePosixPath(entry), included_patterns=self._included_patterns, excluded_patterns=self._excluded_patterns):
```
Comment (tschoonj) [inline rfi_file_monitor/file.py:245]: Here as well

File: rfi_file_monitor/utils/__init__.py [hunk 5]
```diff
@@ -114,7 +114,9 @@ def get_patterns_from_string(input: str, defaults: Optional[List[str]]=None) ->
             return ['*']
     else:
         if input or input.strip():
-             return list(map(lambda x: x.strip(), input.split(','))) + defaults
+             print(defaults)
+             return list(map(lambda x: x.strip(), input.split(','))) + list(defaults)
```
Comment (tschoonj) [inline rfi_file_monitor/utils/__init__.py:118]: This should not be necessary, `defaults` must either `None` or `List[str]`. The cast should happen, if necessary, before the method is called instead.

**Review by tschoonj — CHANGES_REQUESTED** [review]
Hi Laura, many thanks for looking into this! I do have a couple of comments that need to addressed (but not urgently!)

I wish you, Aaron and Sam a merry Christmas! Hope that you will have a great time off, despite us moving into tier 4 on Boxing Day...

File: rfi_file_monitor/engines/directory_watchdog_engine.py (hunk 1 above)
Comment (LAShemilt) [reply rfi_file_monitor/engines/directory_watchdog_engine.py:126]: strange it worked on my Windows machine... I will change it just child...

File: rfi_file_monitor/utils/__init__.py (hunk 5 above)
Comment (LAShemilt) [reply rfi_file_monitor/utils/__init__.py:118]: this threw an error for me... so I needed to cast, but I can cast it before the method is called.

**Review by tschoonj — APPROVED** [review]

**Pull request merged**